components/dashboard/src/configurations/detail/ConfigurationDetailPage.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import type { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import { BreadcrumbNav } from "@podkit/breadcrumbs/BreadcrumbNav";
+import { Button } from "@podkit/buttons/Button";
+import type { UseQueryResult } from "@tanstack/react-query";
+import { Loader2 } from "lucide-react";
+import { useMemo } from "react";
+import Alert from "../../components/Alert";
+import { WidePageWithSubMenu } from "../../components/WidePageWithSubmenu";
+
+export interface PageWithAdminSubMenuProps {
+    children: React.ReactNode;
+    configurationQuery: UseQueryResult<Configuration | undefined, Error>;
+    id: string;
+}
+
+export function ConfigurationDetailPage({ children, configurationQuery, id }: PageWithAdminSubMenuProps) {
+    const { data, error, isLoading, refetch } = configurationQuery;
```

Comment:
This `configurationQuery` business is mostly done so that we don't have to query at two different levels (inner page and outer scaffold), so that we can have only one place where we need to handle errors. The thing is that we also need the data in the children, but I am not aware of an elegant way of passing it to them.

Reply:
Commented on this above, but I'll see if I can think of another pattern here to help make this a bit smoother. 

If we were using a newer version of react-router I think sub routes would be a nice way to solve this. It lets us have our parent route component for `/repositories/:id` - and it would have all of this loading/error handling and the sidebar. We would then have sub-routes defined for each of the pages, and could just pass the configuration object as a prop to them since we'd have it loaded and available.